src/library_syscall.js

```diff
@@ -660,7 +666,9 @@ var SyscallsLibrary = {
   __syscall140: function(which, varargs) { // llseek
     var stream = SYSCALLS.getStreamFromFD(), offset_high = SYSCALLS.get(), offset_low = SYSCALLS.get(), result = SYSCALLS.get(), whence = SYSCALLS.get();
     var offset = offset_low;
+#if WASM_BACKEND == 0
```

Comment:
Fixed in 7fd78fbbb6b2bbb72e59f8e630d647ea290c8b8b

This was due to undefined behavior: asm2wasm was compiling `offset >> 32` to `(i32.const 0)`, s2wasm was compiling it to a no-op, thus it wasn't writing to the vararg buffer that we pass to the syscalls, thus it was getting a garbage value from whatever was on the stack there first.

Fixed it by always just passing `0`.